src/main/java/hachi/Hachi.java

```diff
@@ -0,0 +1,410 @@
+            int indexOfBy = cleanInput.indexOf("/BY") + 3;
+            HachiException.checkDeadlineByDate(indexOfBy);
+
+            String name = line.substring(indexOfName, indexOfBy - 3).trim();
+            String byDate = line.substring(indexOfBy).trim();
+            toAdd = new Deadline(name, byDate);
+        } else {
+            // parse to and from dates here
+            int indexOfName = cleanInput.indexOf("EVENT") + 6;
+            int indexOfStart = cleanInput.indexOf("/FROM") + 5;
+            int indexOfEnd = cleanInput.indexOf("/TO") + 3;
+            HachiException.checkEventDates(indexOfStart, indexOfEnd);
+
+            String name = line.substring(indexOfName, indexOfStart - 5);
+            String fromDate = line.substring(indexOfStart, indexOfEnd - 3).trim();
+            String toDate = line.substring(indexOfEnd).trim();
+            toAdd = new Event(name, fromDate, toDate);
+        }
+
+        taskArrayList.add(toAdd);
+        System.out.println("\tAdded to list: " + toAdd.getName());
+    }
+
+    /**
+     * Function that cleans the user input for mark or unmark requests
+     * and completes the function call as required.
+     *
+     * @param cleanedInput Cleaned input string from user.
+     */
+
+    public static void markOrUnmarkHandler(String cleanedInput) throws HachiException{
+        int numTasks = Task.getTotalNumTasks();
+        HachiException.checkEmptyList(numTasks);
+        int taskNumber = getMarkTaskNumber(cleanedInput);
+        HachiException.checkOutOfBounds(taskNumber);
+        markOrUnmarkTask(taskNumber - 1, !cleanedInput.contains("UNMARK"));
+    }
+
+    /**
+     * Given a task's index and the list of tasks,
+     * mark that task as complete or incomplete.
+     *
+     * @param index Index of the task to be marked.
+     * @param isMark True if task is to be marked as complete, false otherwise
+     */
+
+    public static void markOrUnmarkTask(int index, boolean isMark) {
+        taskArrayList.get(index).setCompleteness(isMark);
+        System.out.println("\tSure, I've done as you requested:");
+
+        Task currentTask = taskArrayList.get(index);
+        String taskType = currentTask.getTaskType();
+        String statusIcon = currentTask.getStatusIcon();
+        System.out.print("\t" + (index + 1) + ": ");
+        System.out.print("[" + taskType + "] ");
+        System.out.print("[" + statusIcon + "] ");
+        System.out.println(currentTask.getName());
+    }
+
+    private static void deleteTask(String cleanedInput) throws HachiException {
+        int numTasks = Task.getTotalNumTasks();
+        HachiException.checkEmptyList(numTasks);
+        int taskNumber = getDeleteTaskNumber(cleanedInput);
+        HachiException.checkOutOfBounds(taskNumber);
+
+        Task toDelete = taskArrayList.get(taskNumber - 1);
+        String taskType = toDelete.getTaskType();
+        String statusIcon = toDelete.getStatusIcon();
```

Comment:
just some coding readability minor issue! (understand the code) but perhaps can name to something like "taskToDelete"